lib/schemas.js

```diff
@@ -34,7 +34,7 @@ Schemas.prototype.resolve = function (id) {
   return this.store[id]
 }
 
-Schemas.prototype.resolveRefs = function (routeSchemas) {
+Schemas.prototype.resolveRefs = function (routeSchemas, dontClearId) {
```

Comment:
Can you please add a bit of description of `dontClearId`, and why somebody should set it to `true`?

Reply:
That boolean needs because it resolve the shared schema when added without removing the `$id` that is needed by `fastify-plugin`.
If I remove that condition (only) this specific test fails:

https://github.com/fastify/fastify/blob/cf7a0963426ff96fc27f4c92bc443f1ddf8f2311/test/decorator.test.js#L635

because the `$id` would be removed from the object, than the same object reference would be given to children that throw a `FST_ERR_SCH_MISSING_ID` error.
I have tried to clone the object, but I get so many fails..

Only `schemas.js` should set it to true

--- 
Context:

When a shared schema is added, a pair of `$id + JSON with $id` is created

When a JSON schema is resolved by the `validation.js` during the startup, the `JSON with $id` is injected in the right place.
During the injection che `$id` field is deleted because, if a user use two or more time the same shared schema AJV will complain for "multiple equals $id".

If a user add a shared schema with a reference to another shared schema and doesn't use it, the shared shema will not resolved and the code break.

Reply:
Can you put some of this explanation in the code itself?

Reply:
Done